data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/ScanObjectWorker.java

```diff
@@ -209,6 +215,17 @@ private void startProcessingObject(final long waitTimeMillis) {
                     }
                     partitionKeys.remove(objectToProcess.get().getPartitionKey());
                 }, ACKNOWLEDGEMENT_SET_TIMEOUT);
+
+                acknowledgementSet.addProgressCheck(
+                        (ratio) -> {
+                            try {
+                                sourceCoordinator.saveProgressStateForPartition(objectToProcess.get().getPartitionKey(), null);
```

Comment:
I think you are saving null to keep holding onto the partition. This has some bleeding in the internal details. I think we should have a dedicated API for it. Something like: `sourceCoordinator.renewPartition(objectToProcess.get().getPartitionKey());` Internally it can save `null`, but this will decouple the details from the caller.

I do see one other similar usage of saving `null`. It would be better to clean this up before it continues to expand.

Reply:
That's fair yeah probably should've been 2 separate methods from the beginning.

Reply:
Actually, can you create bug issue about this so that we don't lose track?

Reply:
Made the issue here (https://github.com/opensearch-project/data-prepper/issues/4928)